landlab/components/concentration_tracker/concentration_tracker_for_diffusion.py

```diff
@@ -0,0 +1,390 @@
+            original parent material concentration (``concentration_in_bedrock``)
+            as it weathers to soil. Use this parameter to differentiate between
+            the concentration in weathered material compared to its parent bedrock.
+        """
+
+        super().__init__(grid)
+        # Store grid and parameters
+
+        # use setters for C_init, C_br, and C_w defined below
+        self.C_init = concentration_initial
+        self.C_br = concentration_in_bedrock
+
+        # get reference to inputs
+        self._soil__depth = self._grid.at_node["soil__depth"]
+        self._soil__depth_old = self._soil__depth.copy()
+        self._soil_prod_rate = self._grid.at_node["soil_production__rate"]
+        self._flux = self._grid.at_link["soil__flux"]
+
+        # create outputs if necessary and get reference.
+        self.initialize_output_fields()
+
+        # Define concentration field (if all zeros, then add C_init)
+        if np.allclose(self._grid.at_node["sediment_property__concentration"], 0.0):
+            self._grid.at_node["sediment_property__concentration"] += self.C_init
+        self._concentration = self._grid.at_node["sediment_property__concentration"]
+
+        if np.allclose(self._grid.at_node["bedrock_property__concentration"], 0.0):
+            self._grid.at_node["bedrock_property__concentration"] += self.C_br
+        self.C_br = self._grid.at_node["bedrock_property__concentration"]
+
+        # use setter for C_w defined below
+        self.C_w = concentration_from_weathering
+
+        # Sediment property concentration field (at links, to calculate dQCdx)
+        self._C_links = np.zeros(self._grid.number_of_links)
+
+        # Sediment property mass field (at links, to calculate dQCdx)
+        self._QC_links = np.zeros(self._grid.number_of_links)
+
+    @property
+    def C_init(self):
+        """Initial concentration in soil/sediment (kg/m^3)."""
+        return self._C_init
+
+    @property
+    def C_br(self):
+        """Concentration in bedrock (kg/m^3)."""
+        return self._C_br
+
+    @property
+    def C_w(self):
+        """Concentration from the weathering process (kg/m^3)."""
+        return self._C_w
+
+    @C_init.setter
+    def C_init(self, new_val):
+        if np.any(new_val < 0.0):
+            raise ValueError("Concentration cannot be negative")
+        self._C_init = return_array_at_node(self._grid, new_val)
+
+    @C_br.setter
+    def C_br(self, new_val):
+        if np.any(new_val < 0.0):
+            raise ValueError("Concentration in bedrock cannot be negative")
+        self._C_br = return_array_at_node(self._grid, new_val)
+
+    @C_w.setter
+    def C_w(self, new_val):
+        if new_val is None:
+            new_val = self._C_br
+        if np.any(new_val < 0.0):
+            raise ValueError("Concentration cannot be negative")
+        self._C_w = new_val
+
+    def concentration(self, dt):
```

Comment:
How about rename this to be something like `calc_concentration`? Using the `calc_` prefix is a convention we've used elsewhere to indication a calculation is being done rather than just returning a value.